### PR TITLE
 Resevered routing names

### LINKED ISSUE #5915: Document that "page" is a reserved route value for Razor Pages
The term `page` (like `controller`, `action`, and `area`) is reserved for routing in Razor Pages. It doesn't look like we have this documented anywhere.

### PR DESCRIPTION
**Pull request by Rick-Anderson**
Fixes #5915

### CODE REVIEW

File: aspnetcore/fundamentals/routing.md [hunk 1]
```diff
@@ -283,7 +283,16 @@ Using a template is generally the simplest approach to routing. Constraints and
 
 Tip: Enable [Logging](xref:fundamentals/logging/index) to see how the built in routing implementations, such as `Route`, match requests.
 
-## Route Constraint Reference
+## Resevered routing names
```
Comment (user135711) [inline aspnetcore/fundamentals/routing.md]: You misspelled  Reserved (**Resevered**)

**Comment by Rick-Anderson** [comment]
@pranavkm 3 minute review

**Comment by pranavkm** [comment]
@rynowak Related to https://github.com/aspnet/Mvc/issues/7876. Think this is valuable to document?

**Comment by rynowak** [comment]
Yes

**Comment by Rick-Anderson** [comment]
@pranavkm 2 minute review

File: aspnetcore/fundamentals/routing.md [hunk 2]
```diff
@@ -283,7 +283,16 @@ Using a template is generally the simplest approach to routing. Constraints and
...
 
-## Route Constraint Reference
+## Reserved routing names
+
+The following keywords are reserved names and can't be used as route names or parameters:
+
+* `page`
+* `controller`
+* `action`
+* `area`
```
Comment (pranavkm) [inline aspnetcore/fundamentals/routing.md]: Also `handler`.

**Comment by pranavkm** [comment]
:shipit:

**Pull request merged**